Foxtrot/Driver/Rewriting/PostRewriteChecker.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Compiler;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Contracts.Foxtrot
+{
+    public sealed class PostRewriteChecker : Inspector
+    {
+        private Member currentMember;
+        private readonly Action<CompilerError> handleError;
+
+        public PostRewriteChecker(Action<CompilerError> handleError)
+        {
+            this.handleError = handleError;
+        }
+
+        public override void VisitProperty(Property property)
+        {
+            bool exchanged = false;
+            if (this.currentMember == null)
+            {
+                this.currentMember = property;
+                exchanged = true;
+            }
+
+            base.VisitProperty(property);
+
+            if (exchanged)
+            {
+                this.currentMember = null;
+            }
+        }
+
+        public override void VisitEvent(Event evnt)
+        {
+            bool exchanged = false;
+            if (this.currentMember == null)
+            {
+                this.currentMember = evnt;
+                exchanged = true;
+            }
+
+            base.VisitEvent(evnt);
+
+            if (exchanged)
+            {
+                this.currentMember = null;
+            }
+        }
+
+        public override void VisitMethod(Method method)
+        {
+            bool exchanged = false;
+            if (this.currentMember == null)
+            {
+                this.currentMember = method;
+                exchanged = true;
+            }
+
+            base.VisitMethod(method);
+
+            if (exchanged)
+            {
+                this.currentMember = null;
+            }
+        }
+
+        public override void VisitMemberBinding(MemberBinding memberBinding)
```

Comment:
This code is not that obvious (as any code that has a bunch of visitors:) ), so I would suggest to add a short description of what problem this code is trying to solve in high level.